src/main/java/pl/byczazagroda/trackexpensesappbackend/config/WebSecurityConfig.java

```diff
@@ -33,6 +33,7 @@ public SecurityFilterChain filterChain(HttpSecurity http) throws Exception {
                 .authorizeHttpRequests(authorize -> authorize
                         .antMatchers("/").permitAll()
                         .antMatchers("/api/auth/login").permitAll()
+                        .antMatchers("/api/auth/refresh").authenticated()
```

Comment:
user should be able to reach refresh endpoint when he is unauthenticated, thats the whole point of refreshing
users token expired -> user refreshes the token -> user is able to continue using api